### PR TITLE
Add Gamer class, improve activities logic.

### PR DESCRIPTION
**Pull request by AlesyaMafa**

### CODE REVIEW

File: src/main/java/com/zharnasek/alesya/game/Gamer.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.zharnasek.alesya.game;
+
+public class Gamer {
+
+    private String name;
+    private Gender gender;
+
+    public enum Gender {
+        MALE, FEMALE;
+    }
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/Gamer.java]: don't create inner classes yet

File: src/main/java/com/zharnasek/alesya/game/PeopleDemo.java [hunk 2]
```diff
@@ -14,22 +14,45 @@ public static void main(String[] args) {
...
+                System.out.print("Input name for him: ");
+                Gamer gamerMale = new Gamer(scanner.next(), Gamer.Gender.MALE);
+                System.out.print("Input name for her: ");
+                Gamer gamerFemale = new Gamer(scanner.next(), Gamer.Gender.FEMALE);
+                String activitiesMale;
+                String activitiesFemale;
+                int randomIndex;
+
+                do {
+                    Random rand = new Random();
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/PeopleDemo.java]: why do you need to create new object random in each cycle?

File: src/main/java/com/zharnasek/alesya/game/PeopleDemo.java [hunk 3]
```diff
@@ -14,22 +14,45 @@ public static void main(String[] args) {
...
+                    do {
+                        randomIndex = rand.nextInt(gameResources.getActivities().size());
+                        activitiesFemale = gameResources.getActivities().get(randomIndex).toString();
+                    } while (!gameResources.getActivities().get(randomIndex).getGender().equals("FEMALE"));
+                    System.out.println(gamerFemale.getName() + ", " + activitiesFemale);
+                    System.out.print("input 'next' to continue, input 'exit' to finish: ");
+                    if (scanner.next().equals("next")) {
+                        do {
+                            randomIndex = rand.nextInt(gameResources.getActivities().size());
+                            activitiesMale = gameResources.getActivities().get(randomIndex).toString();
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/PeopleDemo.java]: DRY

File: src/main/java/com/zharnasek/alesya/game/PeopleDemo.java [hunk 4]
```diff
@@ -14,22 +14,45 @@ public static void main(String[] args) {
...
+                        } while (!gameResources.getActivities().get(randomIndex).getGender().equals("MALE"));
+                        System.out.println(gamerMale.getName() + ", " + activitiesMale);
+                    }
+                    System.out.print("input 'next' to continue, input 'exit' to finish: ");
+                } while (!scanner.next().equals("exit"));
+                flag = true;
             }
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/PeopleDemo.java:45]: I'm crying, not readable, and very complex

**Review by vitali-ivanou — CHANGES_REQUESTED** [review]

File: src/main/java/com/zharnasek/alesya/game/Gamer.java (hunk 1 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/Gamer.java]: updated

File: src/main/java/com/zharnasek/alesya/game/PeopleDemo.java (hunk 2 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/PeopleDemo.java]: updated

File: src/main/java/com/zharnasek/alesya/game/PeopleDemo.java (hunk 3 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/PeopleDemo.java]: updated

File: src/main/java/com/zharnasek/alesya/game/PeopleDemo.java (hunk 4 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/PeopleDemo.java:45]: updated

File: src/main/java/com/zharnasek/alesya/game/Gamer.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public Gender getGender() {
+        return gender;
+    }
+}
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/Gamer.java:55]: Where is equals and hashCode to string?

File: src/main/java/com/zharnasek/alesya/game/RandomActivities.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.zharnasek.alesya.game;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+public class RandomActivities {
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/RandomActivities.java]: bad name

File: src/main/java/com/zharnasek/alesya/game/RandomActivities.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+    public String getRandomActivities(Gamer gamer, GameResource gameResources) {
+        Random rand = new Random();
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/RandomActivities.java]: Why do you pass gamer, do you need several fields from Gamer?

File: src/main/java/com/zharnasek/alesya/game/RandomActivities.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        ArrayList<Activities> list = new ArrayList<>();
+        for (int i = 0; i < gameResources.getActivities().size() - 1; i++) {
+            if (gameResources.getActivities().get(i).getGender().equals(String.valueOf(gamer.getGender()))) {
```
Comment (vitali-ivanou) [inline src/main/java/com/zharnasek/alesya/game/RandomActivities.java]: Need to compare enum, string it's bad for this purpose. Why Gender in activities is String?

**Review by vitali-ivanou — CHANGES_REQUESTED** [review]

File: src/main/java/com/zharnasek/alesya/game/Gamer.java (hunk 5 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/Gamer.java:55]: updated

File: src/main/java/com/zharnasek/alesya/game/RandomActivities.java (hunk 6 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/RandomActivities.java]: updated

File: src/main/java/com/zharnasek/alesya/game/RandomActivities.java (hunk 7 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/RandomActivities.java]: updated

File: src/main/java/com/zharnasek/alesya/game/RandomActivities.java (hunk 8 above)
Comment (AlesyaMafa) [reply src/main/java/com/zharnasek/alesya/game/RandomActivities.java]: updated

**Pull request merged**